db/skipped_sequence.go

```diff
@@ -212,6 +212,56 @@ func binarySearchFunc(a *SkippedSequenceListEntry, seq uint64) int {
 	return -1
 }
 
+// removeSeqRange will remove sequence between x and y from the skipped sequence slice
+func (s *SkippedSequenceSlice) removeSeqRange(startSeq, endSeq uint64) error {
+	s.lock.Lock()
+	defer s.lock.Unlock()
+
+	startIndex, found := s._findSequence(startSeq)
+	if !found {
+		return fmt.Errorf("sequence %d not found in the skipped list", startSeq)
+	}
+	endIndex, found := s._findSequence(endSeq)
```

Comment:
Since we're going to error out right away if the range isn't within a single element, I think it will be more efficient to just perform a single findSequence (on startSeq), and then examine the element to verify that the entire range is within that element.